be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -990,15 +992,23 @@ void DefaultValueColumnIterator::insert_default_data(const TypeInfo* type_info,
 
 Status DefaultValueColumnIterator::next_batch(size_t* n, vectorized::MutableColumnPtr& dst,
                                               bool* has_null) {
+    *has_null = _is_default_value_null;
+    _insert_many_default(dst, *n);
+    return Status::OK();
+}
+
+Status DefaultValueColumnIterator::read_by_rowids(const rowid_t* rowids, const size_t count,
+                                                  vectorized::MutableColumnPtr& dst) {
+    _insert_many_default(dst, count);
+    return Status::OK();
+}
+
+void DefaultValueColumnIterator::_insert_many_default(vectorized::MutableColumnPtr& dst, size_t n) {
     if (_is_default_value_null) {
-        *has_null = true;
-        dst->insert_many_defaults(*n);
+        dst->insert_many_defaults(n);
```

Comment:
has_null is not set?

Reply:
set it at line 995

Reply:
read_by_rowids also use _insert_many_default, should it be set?